plugins/catalog-backend/src/ingestion/processors/CodeOwnersProcessor.ts

```diff
@@ -97,7 +99,8 @@ export async function findRawCodeOwners(
     return data.toString();
   };
 
-  return Promise.any(KNOWN_LOCATIONS.map(readOwnerLocation));
+  const candidates = KNOWN_LOCATIONS.map(readOwnerLocation);
+  return Promise.any(candidates).catch(() => undefined);
```

Comment:
Wondering if we should log the actual error hear for debugging purposes. Unfortunately the processor isn't setup to pass in a logger yet, but I think that would be ideal in this case. Thoughts?

Reply:
So maybe only at debug level then, if none of the candidates find a codeowners file?

I guess it would be valuable to log ACTUAL errors, such as the difference between not finding a file, and failing to fetch them. Even though it's a bit less unlikely given that the entity yaml file was just successfully fetched nearby using the same mechanism, it can happen.

Reply:
@andrewthauer Updated with logging and separate handling of "hard" errors as opposed to 404s. Good idea!

Reply:
Side note: this would have been an excellent place for #2876 . To have been able to emit an event with the right  severity, tagged with the processor id, the current refresh run id, the entity, the location - then it would be super easy to see from the outside what is going on, from several different angles of slicing the events.

Reply:
💯 Another side note is that I've started adding caching and circuit breakers for some the custom catalog processors we are using internally. These particular services call out to other internal services to revolve ids used on annotations. The cache with TTL avoids the need to do the lookup data call. The circuit breaker is used if say the service is down or a bad token, etc. Right now I have these as a per processor concern, but have been toying with the idea of making a higher order catch all breaker that can wrap the preProcess method. Thoughts?

Reply:
Will it be useful to wrap entire processor methods? I mean, for example, a single reader processor may read from many different sources, only one of which may be misbehaving.

What's the cache and cb you are using? Some particular library? We could definitely make recommendations, or make it a core feature to add it to the default url reader and elsewhere where these kinds of transient failures can happen.

Reply:
For caching is using `keyv`, for caching with just in memory for now with a ttl. Using opossum for the circuit breaker. So far I've created a custom cache `getOr` on top of `keyv` which is feed to the CB async method. The two work pretty nicely together, but there is probably room for removing some boiler plate and generalizing even further.

Reply:
That said, your absolutely right. Caching would probably need to be a per processor thing. The CB could be applied to the entire processor as a generic wrapper likely. Which could on theory be an opt in within the backend catalog config. But maybe there is a need to build it into something like the readers. Probably could be controlled through the config as well.

Reply:
I meant it may commonly not be per processor method, but rather per downstream, or even more granular. If github is flaky and can't serve codeowners requests för your open source projects, but your internal bitbucket instance is doing fine, then we should break github requests, not the processor.

Same thing with caching; since it is so use case specific, it will be hard to put at a high level. Again with the codeowners example, in a big monorepo there may be hundreds of different locations/entities in it but they are all covered by one single codeowners file. So caching on the "outside" would essentially cache per location to little effect, while doing it in a dedicated codeowners layer that understands the concept of mapping locations to repo roots, likely will have a good hit rate.

Reply:
@freben - That all makes sense. I agree this will be high dependent on each processor. In my particular cases the processors that require this are very specific so they could be generalized. But yes, there are other cases where it might not be the case. Do you see a need to formalize any of these patterns into some of the core packages? If so, maybe an RFC might make sense?

Reply:
Hadn't thought of it yet, frankly. If you start seeing patterns, and want community (or our) input, then feel free to create one for sure. Whether it solidifies or not, the input and learnings and library choices it provides to others is valuable.